tests/Features/Contexts/FeatureContext.php

```diff
@@ -0,0 +1,216 @@
+<?php
+
+declare(strict_types=1);
+
+/**
+ * Copyright (c) 2013-2018 OpenCFP
+ *
+ * For the full copyright and license information, please view
+ * the LICENSE file that was distributed with this source code.
+ *
+ * @see https://github.com/opencfp/opencfp
+ */
+
+namespace OpenCFP\Test\Features\Contexts;
+
+use Behat\Gherkin\Node\TableNode;
+use Behat\MinkExtension\Context\MinkContext;
+use Behat\Symfony2Extension\Context\KernelDictionary;
+use Illuminate\Database\Capsule;
+use Illuminate\Database\Connection;
+use OpenCFP\Domain\Model\Talk;
+use OpenCFP\Domain\Model\User;
+use OpenCFP\Domain\Services\AccountManagement;
+use Phinx\Console\PhinxApplication;
+use Symfony\Component\Console\Input\StringInput;
+use Symfony\Component\Console\Output\NullOutput;
+
+final class FeatureContext extends MinkContext
+{
+    use KernelDictionary;
+
+    private $accountInfo = [
+        'email'    => 'speaker@example.com',
+        'password' => 'speaker',
+    ];
+
+    /**
+     * @BeforeScenario
+     */
+    public function beforeScenarios()
+    {
+        $_SERVER['CFP_ENV'] = 'testing';
+
+        $this->databaseConnection()->unprepared('DROP DATABASE IF EXISTS cfp_test; CREATE DATABASE cfp_test;');
+        $phinx = new PhinxApplication();
+        $phinx->setAutoExit(false);
+        $phinx->run(new StringInput('migrate'), new NullOutput());
+    }
+
+    /**
+     * @AfterScenario
+     */
+    public function rollbackTransaction()
+    {
+        $this->databaseConnection()->rollBack();
+    }
+
+    private function databaseConnection(): Connection
+    {
+        /** @var Capsule\Manager $capsule */
+        $capsule = $this->getContainer()->get(Capsule\Manager::class);
+
+        return $capsule->getConnection();
+    }
+
+    /**
+     * @Given I am on the signup page
```

Comment:
Maybe we can make these more generic into a 
`@Given i am on the :page page`

Reply:
My preference is to leave the step as-is because it describes a ubiquitous part of the application that *won't* change. The signup page will always be the signup page, but the URL could easily change. Maybe this could be parameterized to  reference a route name instead, but I certainly wouldn't want to tie Gherkin to details like a URL. Gherkin isn't the test, that's the step definition's purpose.